Backend/Remora.Discord.Rest/Handlers/AuthorizationHandler.cs

```diff
@@ -0,0 +1,89 @@
+//
+//  AuthorizationHandler.cs
+//
+//  Author:
+//       Jarl Gullberg <jarl.gullberg@gmail.com>
+//
+//  Copyright (c) 2017 Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Polly;
+using Remora.Discord.Rest.Extensions;
+
+namespace Remora.Discord.Rest.Handlers;
+
+/// <summary>
+/// Represents a delegating handler for adding the Authorization header.
+/// </summary>
+internal class AuthorizationHandler : DelegatingHandler
```

Comment:
`BotTokenAuthorizationHandler` might be a more appropriate name for this type.

Reply:
I would actually prefer `TokenAuthorizationHandler`.
Also we might should add the TokenType to the ITokenStore, so that we might add also Bearer Token support if one wants to allow it to be used in OAuth context.

Reply:
Sounds good to me!